src/simply/simply_stage.c

```diff
@@ -200,6 +224,26 @@ static void circle_element_draw(GContext *ctx, SimplyStage *self, SimplyElementC
   }
 }
 
+static void radial_element_draw(GContext *ctx, SimplyStage *self, SimplyElementRadial *element) {
+  // element->border_color
+  // element->border_width
+  if (element->border_color.a) {
+    printf("Draw Arc");
+    GRect frame = grect_inset(element->frame, GEdgeInsets(0));
+    graphics_context_set_fill_color(ctx, element->background_color);
+    graphics_draw_arc(ctx, frame, GOvalScaleModeFitCircle,
+                        DEG_TO_TRIGANGLE(element->angle_start),
+                        DEG_TO_TRIGANGLE(element->angle_end));
+  } else {
+    printf("Draw Radial");
+    GRect frame = grect_inset(element->frame, GEdgeInsets(0));
+    graphics_context_set_fill_color(ctx, element->background_color);
+    graphics_fill_radial(ctx, frame, GOvalScaleModeFitCircle, element->radius,
+                        DEG_TO_TRIGANGLE(element->angle_start),
+                        DEG_TO_TRIGANGLE(element->angle_end));
+  }
+}
```

Comment:
Ah, it seems you've actually did most of the work! I think you can complete the implementation with some changes.

  1. The element should be able to draw both the background and border, similar to an html element, simply remove the `else`. That is, draw arc is inside `if (element->border_color.a) { ... }` and fill radial is inside a new `if (element->background_color.a) { ... }`.
  2. The border should be drawn last so it appears above the background, that way the background won't hide larger stroke widths (users can use two Radials if they want the background to draw above). Simply swap the if statements around.
  3. You can just do `GRect frame = element->frame` once at the top of the function instead of inside each if branch since it is the same for both.
  4. Make sure DEG_TO_TRIANGLE lines up with the left `(` parenthesis of the call site it belongs to. That is, the `c` in `ctx` will line up with the `D` in `DEG_TO_TRIANGLE`. [This is Google C++ function call style](https://google.github.io/styleguide/cppguide.html#Function_Calls).
  5. Finally, for draw arc, replace the `set_fill_color` with `graphics_context_set_stroke_color(ctx, element->border_color)` and `graphics_context_set_stroke_width(ctx, element->border_width)` right before drawing. The other element draw functions will also need to set the stroke width too so the stroke width doesn't leak throughout.